code-studio/src/js/components/BeeCellEditor.jsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * @overview React component to allow for easy editing and creation of
+ * BeeCells
+ * @see @cdo/apps/maze/beeCell
+ */
+/* global React */
+
+var BeeCellEditor = module.exports = React.createClass({
+  propTypes: {
+    cell: React.PropTypes.object.isRequired,
+    row: React.PropTypes.number.isRequired,
+    col: React.PropTypes.number.isRequired,
+    onUpdate: React.PropTypes.func.isRequired,
+  },
+
+  handleChange: function (event) {
+    var values = {};
+    var nodes = this.getDOMNode().querySelectorAll('[name]');
+    for (var i = 0, node; (node = nodes[i]); i++) {
+      values[node.name] = isNaN(node.value) ? undefined : parseInt(node.value);
+    }
+    this.props.onUpdate(values);
+  },
+
+  /**
+   * Focus and select the input that contains this cell's serialized
+   * JSON representation on creation or update, to provide a convenient
+   * copy/paste update route.
+   */
+
+  render: function () {
+    var values = this.props.cell.serialize();
+
+    // If the cell is a variable cloud, its feature MUST be variable
+    if (this.props.cell.isVariableCloud()) {
+      values.featureType = 2;
```

Comment:
Would it make sense to use BeeCell.FeatureType.VARIABLE here? (as well as in the < option /> elements)